context.go

```diff
@@ -1417,6 +1465,43 @@ func (c *Context) addInterVariantDependency(origModule *moduleInfo, tag Dependen
 	atomic.AddUint32(&c.depsModified, 1)
 }
 
+// findBlueprintDescendants returns a map linking parent Blueprints files to child Blueprints files
+// For example, if paths = []string{"a/b/c/Android.bp", "a/Blueprints"},
+// then descendants = {"":[]string{"a/Blueprints"}, "a/Blueprints":[]string{"a/b/c/Android.bp"}}
+func findBlueprintDescendants(paths []string) (descendants map[string][]string, err error) {
+	// make mapping from dir path to file path
+	filesByDir := make(map[string]string, len(paths))
+	for _, path := range paths {
+		dir := filepath.Clean(filepath.Dir(path))
+		_, alreadyFound := filesByDir[dir]
+		if alreadyFound {
+			return nil, fmt.Errorf("Found two Blueprint files in directory %v : %v and %v", dir, filesByDir[dir], path)
+		}
+		filesByDir[dir] = path
+	}
+	// generate the descendants map
+	descendants = make(map[string][]string, len(filesByDir))
+	for childDir, childFile := range filesByDir {
+		prevAncestorDir := childDir
+		for {
```

Comment:
Not sure if its better, but this could be:
for ancestorDir := filepath.Dir(prevAncestorDir); ancestorDir != prevAncestorDir {

Reply:
It does also need to add childFile as a descendant of descendants[""] if it reaches the root directory. Could you clarify on the syntax that you would like, or shall we just leave it?